src/types.jl

```diff
@@ -35,15 +35,15 @@ abstract type Records end
 
 # Create a instance of a `Records` subtype, with all fields (assumed to be Vector)
 # containing `nrow` elements (initialised to undefined values, as they'll be overwritten).
-(::Type{R})(nrow) where {R <: Records} = R(map(T -> T(undef, nrow), fieldtypes(R))...)
+(::Type{R})(nrow) where {R <: Records} = R(map(T -> T(undef, (nrow,)), fieldtypes(R))...)
 
 # Store data in column table so conversion to DataFrame efficient.
 Tables.istable(::Type{<:Records}) = true
 Tables.columnaccess(::Type{<:Records}) = true
 Tables.columns(x::Records) = x
 Tables.getcolumn(x::Records, i::Int) = getfield(x, i)
 Tables.columnnames(R::Type{<:Records}) = fieldnames(R)
-Tables.schema(R::Type{<:Records}) = Tables.Schema(fieldnames(R), fieldtypes(R))
+Tables.schema(x::R) where {R <: Records} = Tables.Schema(fieldnames(R), fieldtypes(R))
```

Comment:
since we have to define `schema` on `Transformers` objects (not the type itself) we define this on objects too, so we can generically call `schema(x)` (e.g. in `show`)